src/testrender/optixraytracer.cpp

```diff
@@ -844,6 +850,26 @@ OptixRaytracer::build_accel()
 
 
 
+void
+OptixRaytracer::prepare_background()
+{
+    if (getBackgroundShaderID() >= 0) {
+        const int bg_res = std::max<int>(32, getBackgroundResolution());
+        CUDA_CHECK(cudaMalloc(reinterpret_cast<void**>(&d_bg_values),
```

Comment:
Would there be a way to wrap the cudaMalloc calls so that user code doesn't need to have as many `reinterpret_casts` everywhere ? It could also take care of calling `m_ptrs_to_free.push_back()` at the same time to avoid mistakes.

Reply:
Ah, there is already an unused function that does almost exactly that. I'll look at adapting it to streamline these operations.

Reply:
I think it would be a nice improvement, but maybe a little out-of-scope for this already big change.

Reply:
I just took a stab at this and it makes the code a bit easier on the eyes. How do you feel about this?
```c++
#define DEVICE_ALLOC(size) reinterpret_cast<CUdeviceptr>(device_alloc(size))
#define COPY_TO_DEVICE(dst_device, src_host, size) \
    copy_to_device(reinterpret_cast<void*>(dst_device), src_host, size)
```
It's a wrapper for the wrapper that takes care of some of the casting. I don't necessarily want to completely obscure the fact that we're dealing the `CUdeviceptr` and not `void*`, although they are for the most part interchangeable.